doc/contribute/doc-guidelines.rst

```diff
@@ -0,0 +1,308 @@
+   ###############################################
+
+
+Other .rst documents can link to this document using the ``:ref:`doc_guidelines``` tag and
+it will show up as :ref:`doc_guidelines`.  This type of internal cross reference works across
+multiple files, and the linking text is obtained from the document source so if the title changes,
+the link text will update as well.
+
+
+Non-ASCII Characters
+********************
+
+You can insert non-ASCII characters such as a Trademark symbol, use the notation ``|trade|``.
+These replacement names are defined in an include file used during the sphinx processing
+of the reST files.  The names of these replacement characters are the same as used in HTML
+entities used to insert characters in html, e.g., \&trade; and are defined in the
+file ``sphinx_build/substitutions.txt`` and listed here:
+
+.. literalinclude:: ../substitutions.txt
+   :language: rst
+
+We've kept the substitutions list small but others can be added as
+needed by submitting a change to the ``substitutions.txt`` file.
+
+Code and Command Examples
+*************************
+
+Use the ReST ``code-block`` directive to create a highlighted block of
+fix-width text, typically used for showing formatted code or console
+commands and output.  Smart syntax highlighting is also supported (using the
+Pygments package) but you can also specify the highlighting language.
+For example::
+
+   .. code-block:: c
+
+      struct _k_object {
+         char *name;
+         u8_t perms[CONFIG_MAX_THREAD_BYTES];
+         u8_t type;
+         u8_t flags;
+         u32_t data;
+      } __packed;
+
+Note the blank line between the ``code-block`` directive and the first
+line of the code-block body, and the body content is indented three
+spaces relative to the current text indent level.  (If a ``code-block``
+were part of a bullet list's content, the directive would be indented to
+the same level as the text of the bullet list's contents.)
+
+You can specify other languages for the ``code-block`` directive,
+including ``c``, ``python``, and ``rst``, but also ``console``,
+``bash``, or ``shell``. If you want no syntax highlighting, use the
+language ``none``,  for example::
+
+   .. code-block:: none
+
+      This would be a block of text styled with a background
+      and box, but with no syntax highlighting.
+
+Would display as:
+
+
+   .. code-block:: none
+
+      This would be a block of text styled with a background
+      and box, but with no syntax highlighting.
+
+There's a shorthand for writing code blocks too; end the introductory
+paragraph with a double colon (``::``) and indent the code block content
+by three spaces.  On output, only one colon will be shown.
+
+Tabs, spaces, and indenting
+***************************
+
+Indenting is significant in ReST file content, and using spaces is
+prefered.  Extra indenting can (unintentionally) change the way content
+is rendered too.  In general, use three spaces for each indent level,
```

Comment:
I think the real rule here is to indent to the first non-whitespace character. Using three spaces for list items would be a bit unnatural and not work well with most people's text editors.

Reply:
yep